### PR TITLE
[circle-execution-plan] Support execution plan for onert-micro

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This PR adds support for onert-micro in circle-execution-plan.

for issue: https://github.com/Samsung/ONE/issues/9849

draft: https://github.com/Samsung/ONE/pull/9893

ONE-DCO-1.0-Signed-off-by: Artem Balyshev a.balyshev@partner.samsung.com

### CODE REVIEW

**Comment by seanshpark** [comment]
Q) this change is in `compiler`. Title says `for onert-micro`. are compiler modules shared with `onert-micro` ?

**Comment by BalyshevArtem** [comment]
> Q) this change is in `compiler`. Title says `for onert-micro`. are compiler modules shared with `onert-micro` ?

These changes are necessary so that the `circle-execution-planner` can record the result execution plan to metadata for further reading and execution in `onert-micro`, due to we want to support static allocations in `onert-micro`, so we need to support it also in `circle-execution-planner`.

File: compiler/circle-execution-plan/src/CircleExecutionPlan.cpp [hunk 1]
```diff
@@ -36,6 +36,29 @@ int entry(int argc, char **argv)
...
   arser.add_argument("--platform").default_value("linux").help("Platform name: linux mcu cmsisnn");
+  arser.add_argument("--allocating_mode")
+    .default_value("common")
+    .help("Buffer type name (only onert-micro option):"
+          "common - a single buffer is considered for all allocations"
+          "split - there are three buffers: for input,"
+          " for output and for intermediate tensors");
+  arser.add_argument("--runtime")
+    .default_value("onert_micro")
+    .help("Target runtime name: luci-interpreter onert-micro");
```
Comment (binarman) [inline compiler/circle-execution-plan/src/CircleExecutionPlan.cpp:47]: This code do not contain handler of `luci-interpreter` runtime.
Do you plan to add it later or this line is obsolete?

Comment (BalyshevArtem) [reply compiler/circle-execution-plan/src/CircleExecutionPlan.cpp:47]: yes, it will be added in next PR

**Review by binarman — APPROVED** [review]
LGTM

**Pull request merged**